source/reference/menus.rst

```diff
@@ -0,0 +1,107 @@
+===================
+ Menus - Reference
+===================
+
+.. seealso::
+
+   :doc:`Documentation on menus </customization/menus>`
+      Explains how menus work and what you can do
+
+
+File Format
+===========
+
+=============  ===========================================
+**Format**     JSON (with comments)
+
+**Extension**  ``.sublime-menu``
+
+**Name**       One out of the list of available menus.
+               See :ref:`menu-types` for the complete name list
+               and what menu they represent.
+
+**Location**   Any
+
+**Content**    A list of :ref:`menu-item-objects`.
+=============  ===========================================
+
+
+Example
+*******
+
+The following is an excerpt
+from the default :file:`Main.sublime-menu` file.
+
+.. code-block:: json
+   :linenos:
+
+   [
+       {
+           "caption": "Edit",
+           "mnemonic": "E",
+           "id": "edit",
+           "children":
+           [
+               { "command": "undo", "mnemonic": "U" },
+               { "command": "redo_or_repeat", "mnemonic": "R" },
+               {
+                   "caption": "Undo Selection",
+                   "children":
+                   [
+                       { "command": "soft_undo" },
+                       { "command": "soft_redo" }
+                   ]
+               },
+               { "caption": "-", "id": "clipboard" },
+               { "command": "copy", "mnemonic": "C" },
+               { "command": "cut", "mnemonic": "t" },
+               { "command": "paste", "mnemonic": "P" },
+               { "command": "paste_and_indent", "mnemonic": "I" },
+               { "command": "paste_from_history", "caption": "Paste from History" }
+           ]
+       }
+   ]
+
+
+.. _menu-item-objects:
+
+"Menu Item" objects
+===================
+
+:ref:`Menu items <menu-items>` may have the following properties.
+
+Unless you are referencing an existing item via ID,
+each menu item must define either
+``children``, ``command`` or ``caption``.
+
+``command``
+   Name of the command to be called
+   when the menu item is selected.
+
+``args``
+   Object of arguments to the command.
```

Comment:
I don't think we ever say "object" in this case in other places. Perhaps:

    Arguments ... as a JSON object.